backtrace-library/src/main/java/backtraceio/library/models/json/ReportDataBuilder.java

```diff
@@ -0,0 +1,40 @@
+package backtraceio.library.models.json;
+import java.util.HashMap;
+import java.util.Map;
+
+import backtraceio.library.common.TypeHelper;
+import backtraceio.library.models.Tuple;
+
+public class ReportDataBuilder {
+
+    /**
+     * Divide custom user attributes into primitive and complex attributes and add to this object
+     *
+     * @param attributes client's attributes
+     */
+    public static Tuple<Map<String, String>, Map<String, Object>> getReportAttribues(Map<String, Object> attributes) {
+        HashMap<String, String> reportAttributes = new HashMap<>();
+        HashMap<String, Object> reportAnnotations = new HashMap<>();
+
+        if(attributes == null) {
+            return new Tuple<>(reportAttributes, reportAnnotations);
+        }
+
+        for (Map.Entry<String, Object> entry : attributes.entrySet()) {
+            Object value = entry.getValue();
+            if (value == null) {
+                reportAttributes.put(entry.getKey(), "null");
```

Comment:
why we need to have string 'null' instead of normal null?

Reply:
Null values will be refused by the metrics system

Reply:
sure - this is also interesting case for the future. In the past we were dropping attributes with nullable value

Reply:
Changing the implementation so this will be more obvious. Thanks for the suggestion

Reply:
Changed the implementation. I still think nullable values are important for reports (we shouldn't skip them) however for metrics, we cannot send them to API. I extended the interface to make sure we have a flexible option to use them